backend/services/kakao_pay_service.py

```
@@ -0,0 +1,255 @@
+                    ios_app_scheme=response_data.get("ios_app_scheme"),
+                    created_at=datetime.now()
+                )
+                
+        except PaymentErrorResponse:
+            raise
+        except Exception as e:
+            print(f"Payment preparation error: {e}")
+            raise PaymentErrorResponse(
+                error_code="PREPARATION_ERROR",
+                error_message=f"Failed to prepare payment: {str(e)}"
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)